Source/Core/DolphinTool/ExtractCommand.cpp

```diff
@@ -0,0 +1,313 @@
+  std::unique_ptr<DiscIO::FileInfo> info = filesystem->FindFileInfo(path);
+  return info;
+}
+
+static int VolumeSupported(const DiscIO::Volume& disc_volume)
+{
+  switch (disc_volume.GetVolumeType())
+  {
+  case DiscIO::Platform::WiiWAD:
+    fmt::println(std::cerr, "Error: Wii WADs are not supported.");
+    return false;
+  case DiscIO::Platform::ELFOrDOL:
+    fmt::println(std::cerr,
+                 "Error: *.elf or *.dol have no filesystem and are therefore not supported.");
+    return false;
+  case DiscIO::Platform::WiiDisc:
+  case DiscIO::Platform::GameCubeDisc:
+    return true;
+  default:
+    fmt::println(std::cerr, "Error: Unknown volume type.");
+    return false;
+  }
+}
+
+static void ExtractDirectory(const DiscIO::Volume& disc_volume, const DiscIO::Partition& partition,
+                             const std::string& path, const std::string& out, bool mute)
+{
+  const DiscIO::FileSystem* filesystem = disc_volume.GetFileSystem(partition);
+  if (!filesystem)
+    return;
+
+  const std::unique_ptr<DiscIO::FileInfo> info = filesystem->FindFileInfo(path);
+  u32 size = info->GetTotalChildren();
+  u32 files = 0;
+  ExportDirectory(
+      disc_volume, partition, *info, true, path, out,
+      [&files, &size, &mute](const std::string& current) {
+        files++;
+        const float progress = static_cast<float>(files) / static_cast<float>(size) * 100;
+        if (!mute)
+          fmt::println(std::cerr, "Extracting: {} | {}%", current, static_cast<int>(progress));
+        return false;
+      });
+}
+
+static bool ExtractSystemData(const DiscIO::Volume& disc_volume, const DiscIO::Partition& partition,
+                              const std::string& out)
+{
+  return ExportSystemData(disc_volume, partition, out);
+}
+
+static void ExtractPartition(const DiscIO::Volume& disc_volume, const DiscIO::Partition& partition,
+                             const std::string& out, bool mute)
+{
+  ExtractDirectory(disc_volume, partition, "", out + "/files", mute);
+  ExtractSystemData(disc_volume, partition, out);
+}
+
+static void ListVolume(const DiscIO::Volume& disc_volume, const std::string& path,
+                       std::string& specific_partition_name)
+{
+  std::ofstream output_file;
+  output_file.open("ls.txt");
+
+  for (DiscIO::Partition& p : disc_volume.GetPartitions())
+  {
+    const std::optional<u32> partition_type = disc_volume.GetPartitionType(p);
+    if (!partition_type)
+    {
+      fmt::println(std::cerr, "Error: Could not get partition type.");
+      return;
+    }
+    const std::string partition_name = DiscIO::NameForPartitionType(*partition_type, true);
+
+    if (!specific_partition_name.empty() &&
+        !Common::CaseInsensitiveEquals(partition_name, specific_partition_name))
+      continue;
+
+    const std::string partition_start =
+        fmt::format("/// PARTITION: {} <{}> ///\n", partition_name, path);
+    fmt::print(std::cerr, "{}", partition_start);
```

Comment:
Well, without the automatic file output, I guess you could just skip the entire procedure since you won't get any output either way. Outputting an error for having the quiet flag defined is a little weird imo, but I guess that is what I'll make it do.